rllib/offline/offline_data.py

```diff
@@ -93,7 +95,7 @@ def sample(
             # sampling later with a different batch size would need a
             # reinstantiation of the iterator.
             self.batch_iterator = self.data.map_batches(
-                OfflinePreLearner,
+                self.prelearner_class,
```

Comment:
Ah, basically, this is where the actual choice for the prelearner_class to be used should be made based on the user's config.

So move the logic from BCConfig/MARWILConfig to here.

Reply:
This would mean we have to define for each algorithm a default prelearner class in the config. Was this your idea?